packages/lib/server/repository/domainWideDelegation.ts

```diff
@@ -0,0 +1,203 @@
+import type { Prisma } from "@prisma/client";
+import z from "zod";
+
+import type { IDomainWideDelegationRepository } from "@calcom/features/domain-wide-delegation/domain-wide-delegation-repository.interface";
+import { getFeatureFlag } from "@calcom/features/flags/server/utils";
+import logger from "@calcom/lib/logger";
+import { prisma } from "@calcom/prisma";
+
+import { OrganizationRepository } from "./organization";
+
+const serviceAccountKeySchema = z
+  .object({
+    client_email: z.string(),
+    private_key: z.string(),
+    client_id: z.string(),
+  })
+  .passthrough();
+
+export type ServiceAccountKey = z.infer<typeof serviceAccountKeySchema>;
+
+const repositoryLogger = logger.getSubLogger({ prefix: ["DomainWideDelegationRepository"] });
+const domainWideDelegationSafeSelect = {
+  id: true,
+  enabled: true,
+  domain: true,
+  createdAt: true,
+  updatedAt: true,
+  organizationId: true,
+  workspacePlatform: {
+    select: {
+      name: true,
+      slug: true,
+    },
+  },
+};
+
+const domainWideDelegationSelectIncludesServiceAccountKey = {
+  ...domainWideDelegationSafeSelect,
+  serviceAccountKey: true,
+};
+
+export class DomainWideDelegationRepository implements IDomainWideDelegationRepository {
```

Comment:
Why do we need an interface for this?

Reply:
Just to make sure if we add a new method to this repository we also add that method to mock repository

Reply:
But why do we need MockRepository?

Reply:
It's just cleaner way to not expose the methods of domainWideDelegation when it is disabled. similar to packages/features/calendar-cache/calendar-cache.ts

Reply:
Okay, just went into detail and noticed it is a pattern introduced in Calendar Cache webhook PR. We should be fine for now but I will dig deep if we can avoid maintaining these mock files and interface files later.